opuscleaner/filters/normalize_whitespace.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+"""Normalize whitespace filter.
+
+Strips all leading and trailing whitespaces. Optionally, collapses all groups
+of whitespaces into a single space
+
+"""
+import argparse
+import sys
+
+
+def collapse_whitespace(s):
+    """Collapses whitespace groups into a single space."""
+    return " ".join(s.split())
+
+
+def clean(collapse):
+    """Runs the filter."""
+
+    for line in sys.stdin:
+        fields = line.strip().split("\t")
```

Comment:
If the first or last column is empty, this will also throw out the column, producing a line without `\t`.

```py
fields = line.strip("\r\n").split("\t")
```
seems safer.

Reply:
Oh yeah you're right that second strip will also get rid of the line ending that the last column will have.